src/main/java/edu/kit/datamanager/repo/util/validators/impl/ISBNValidator.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2021 Karlsruhe Institute of Technology.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.kit.datamanager.repo.util.validators.impl;
+
+import edu.kit.datamanager.exceptions.BadArgumentException;
+import edu.kit.datamanager.repo.util.validators.IIdentifierValidator;
+import org.datacite.schema.kernel_4.RelatedIdentifierType;
+
+/**
+ * This class validates ISBN10 and ISBN13.
+ * @author maximilianiKIT
+ */
+public class ISBNValidator implements IIdentifierValidator {
+
+    @Override
+    public RelatedIdentifierType getSupportedType() {
+        return RelatedIdentifierType.ISBN;
+    }
+
+    @Override
+    public boolean isValid(String input) {
+        if (input.length() == 10) return isValidISBN10(input);
+        else if (input.length() == 13) return isValidISBN13(input);
+        LOG.error("Invalid input length. Please use a valid ISBN10 or ISBN13.");
+        throw new BadArgumentException("Invalid input length. Please use a valid ISBN10 or ISBN13.");
+    }
+
+    /**
+     * This method validates ISBN-13
+     * @param input ISBN13 to validate
+     * @return true if valid
+     * Throws a BadArgumentException if the input is invalid.
+     */
+    private boolean isValidISBN13(String input) {
+        int sum = 0;
+        boolean isOne = true;
+        for (int i = 0; i < 13; i++) {
+            int number = input.charAt(i);
+            if (!isOne) number *= 3;
+            sum += number;
+            isOne = !isOne;
+        }
+        if ((sum % 10) == 0){
+            LOG.info("Valid ISBN13: {}", input);
+            return true;
+        } else{
+            LOG.error("Invalid input: {}", input);
+            throw new BadArgumentException("Invalid input");
```

Comment:
Please provide the input!
String message = String.format("Invalid ISBN number: '%s'", input)
LOGGER.error(message);
throw ... (message);